src/main/kotlin/no/nav/familie/tilbake/behandling/BehandlingMapper.kt

```diff
@@ -0,0 +1,78 @@
+package no.nav.familie.tilbake.behandling
+
+import no.nav.familie.kontrakter.felles.tilbakekreving.OpprettTilbakekrevingRequest
+import no.nav.familie.kontrakter.felles.tilbakekreving.VergeType
+import no.nav.familie.tilbake.behandling.domain.Behandling
+import no.nav.familie.tilbake.behandling.domain.Behandlingstype
+import no.nav.familie.tilbake.behandling.domain.EksternBehandling
+import no.nav.familie.tilbake.behandling.domain.Fagsak
+import no.nav.familie.tilbake.common.ContextService
+import no.nav.familie.tilbake.varsel.Varsel
+import no.nav.familie.tilbake.varsel.Varselsperiode
+import no.nav.familie.tilbake.verge.Verge
+import no.nav.familie.tilbake.verge.Vergetype
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import java.time.LocalDate
+
+object BehandlingMapper {
+
+    private val logger: Logger = LoggerFactory.getLogger(this.javaClass)
+    fun tilDomeneBehandling(opprettTilbakekrevingRequest: OpprettTilbakekrevingRequest,
+                            fagsystem: String,
+                            fagsak: Fagsak): Behandling {
+        val eksternBehandling = EksternBehandling(eksternId = opprettTilbakekrevingRequest.eksternId)
+        val varsel = tilDomeneVarsel(opprettTilbakekrevingRequest)
+        val verger = tilDomeneVerge(fagsystem, opprettTilbakekrevingRequest)
+
+        return Behandling(fagsakId = fagsak.id, type = Behandlingstype.TILBAKEKREVING,
+                          ansvarligSaksbehandler = ContextService.hentSaksbehandler(),
+                          behandlendeEnhet = opprettTilbakekrevingRequest.enhetId,
+                          behandlendeEnhetsNavn = opprettTilbakekrevingRequest.enhetsnavn,
+                          manueltOpprettet = opprettTilbakekrevingRequest.manueltOpprettet,
+                          eksternBehandling = setOf(eksternBehandling),
+                          varsler = setOf(varsel),
+                          verger = verger)
+    }
+
+    private fun tilDomeneVarsel(opprettTilbakekrevingRequest: OpprettTilbakekrevingRequest): Varsel {
+        val varselsperioder = HashSet<Varselsperiode>()
+        for (periode in opprettTilbakekrevingRequest.feilutbetaltePerioder.perioder) {
+            varselsperioder.add(Varselsperiode(fom = periode.fom, tom = periode.tom))
+        }
```

Comment:
val varselsperioder =
                opprettTilbakekrevingRequest.feilutbetaltePerioder.perioder.map {
                    Varselsperiode(fom = it.fom, tom = it.tom)
                }

Reply:
fikset, takk for den 👍